main/2024/CG-06.md

```diff
@@ -93,10 +93,1859 @@ scheduled times if previous sessions end early.
+
+TL: I think so far with this early stage of this proposal, it’s already proven as a well lit path. We’ve had users following the same path with WebGPU with web builtins that will go faster. Technically this proposal sets a path that scales a lot better than trying to add all of this to the core spec.
+
+RH: Hopefully when we write the spec, the spec infrastructure will make it easy to extend.
+
+SC: On the table mutability, immutable tables would be great.
+
+MF: I’m sorry about this bike shed topic. How tied are you to the cute name for imports [the cute “‘“ namespace]? I would prefer to set a precedent for using the "wasm:" prefix for built-in modules.
+
+RH: I agree, that would be my preference, but in the binary format the import namespace is repeated every single time. What would make it feasible is if we had a way of deduplicating names in the binary level, e.g. through a new name section. If we had that, we could do something better. Otherwise, the single character is the best we can do.
+
+MF: Do we need to do it in that order?
+
+RH: Practically I don’t think anyone will use this if we do it in that order.
+
+NF: Does that code size actually matter given that you are probably compressing over the wire?
+
+RH: Question for Google Sheets more than me. I’d guess it does. I can open an issue to revisit that before phase 4.
+
+TL: One idea we’ve experimented with is that you can have a configurable name for the string constant namespace. That would remove the cuteness as well. We experimented with that and when you configured it to just be the quote character, it was 2% slower. With the "string-constants" name, it was 12% slower. It’s significantly more complicated to make it configurable in the engine so keeping it this way as proposed.
+
+BT: Why not the empty string?
+
+RH: Yeah I like that.
+
+CW: Now I’m thinking about this new global trick…it’s interesting because previously we could understand it as a special case of compile time imports, now it’s compile time imports + import name reflection. One concrete question: for normal wasm modules, the import things will be quite small, so are there implementation defined limits for import names that will clash with these string constants?
+
+RH: I’m sure we have one, feels familiar. But haven’t heard from anyone experimenting with it that it’s been an issue, and it’s something we could easily increase.
+
+BT: Just to prompt your imagination a bit. Instead of a quote, just a json literal. The engine has amazing optimization for json and would create a literal expediently.
+
+RH: I will file an issue, you’re all invited to comment. Phase 3 poll?
+
+TL: Poll to advance to phase 3!
+
+|SF|F|N|A|SA|
+|--|-|-|-|--|
+|17|14|12|0|0
+
+TL: That is consensus.
+
+Room: *applause*
+
+#### Memory control (Deepti Gandluri and Ben Visness)
+
+DG & BV presenting slides (TODO: link to slides); notes not already in slides:
```

Comment:
@dtig @bvisness do you have a slide link?

Reply:
Here: https://docs.google.com/presentation/d/1n89xOMpZMpic9MBIuR3mkxABQCsYG6U7lr33TdWlVpQ/edit?usp=sharing